# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/clients/manager/ClientChannelHelper.h
# Commit: 6125bb02952a
# Full Hash: 6125bb02952aff71c0d757e238a7752101b2d6c2
# Author: Jean-Yves Avenard <jyavenard@mozilla.com>
# Date: 2019-09-03 04:06:20
# Regressor Bug: 1556489
# File Overlap Count: 1
# Description:
#   Bug 1556489 - P23 - Only create new ClientSource objects in the content, but use the recreated info from redirects in the parent. r=asuth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D42260
# ==============================================================================

diff -r 06529d12b589 -r 6125bb02952a dom/clients/manager/ClientChannelHelper.h
--- a/dom/clients/manager/ClientChannelHelper.h	Mon Aug 26 12:18:57 2019 +1000
+++ b/dom/clients/manager/ClientChannelHelper.h	Tue Aug 27 08:53:37 2019 +1000
@@ -22,10 +22,15 @@
 // properly handle creating a new ClientSource on cross-origin redirect
 // and propagate the current reserved/initial client on same-origin
 // redirect.
+// Pass true for aManagedInParent if redirects will be handled in the
+// parent process (by a channel with AddClientChannelHelperInParent),
+// and this process only sees a single switch to the final channel,
+// as done by DocumentChannel.
 nsresult AddClientChannelHelper(nsIChannel* aChannel,
                                 Maybe<ClientInfo>&& aReservedClientInfo,
                                 Maybe<ClientInfo>&& aInitialClientInfo,
-                                nsISerialEventTarget* aEventTarget);
+                                nsISerialEventTarget* aEventTarget,
+                                bool aManagedInParent);
 
 nsresult AddClientChannelHelperInParent(nsIChannel* aChannel,
                                         nsISerialEventTarget* aEventTarget);